pyiron/sphinx/base.py

```diff
@@ -1515,6 +1566,107 @@ def write_spins_constraints(self, file_name="spins.in", cwd=None, spins_str=None
             s.logger.debug("No magnetic moments")
 
 
+class Group(dict):
```

Comment:
While this class is currently Sphinx specific it might make sense to move it to https://github.com/pyiron/pyiron/blob/master/pyiron/base/generic/parameters.py to make it available for other codes as well. In addition it might be reasonable to think about a slightly more specific name of this class like InputGroup . And I noticed that quite a bit of the functionality implemented here is already available in https://github.com/cdgriffith/Box so it might even be possible to derive this class from there.

Reply:
I suggest to postpone this generalization to a new issue when you have the first example (LAMMPS by @pmrv  ?)

Reply:
@jan-janssen I also saw the Box object. `Group` already has the main features we need, but there are indeed some nice extras that might be nice to have with Box. My only hesitation was adding it to the installation requirements. But if you're okay with it, I'm all for it.

I also think we should call them `InputGroup`s, and I'm happy to leave this to @pmrv's generalization.